src/odemis/driver/hamamatsurx.py

```diff
@@ -0,0 +1,2043 @@
+                        #     try:
+                        #         additional_info.append(self._commandport.recv(4096))  # receive more data
+                        #     except socket.timeout:
+                        #         break
+
+                        additional_info = ""
+                        timeout = 1
+                        start = time.time()
+                        time_now = time.time()
+                        while time_now < start + timeout:
+                            try:
+                                # continue listening as there is additional info in coming
+                                additional_info += self._commandport.recv(4096)   # receive more data
+                                time_now = time.time()
+                            except Exception:
+                                break
+                        try:
+                            additional_info = additional_info.split("\r")[:-1]
+                            for i, item in enumerate(additional_info):
+                                rargs.append(item.replace("\n", ""))
+                            msg_splitted[2] = rargs
+                        except Exception:
+                            logging.error("Could not retrieve readout camera information.")
+                        self._getReadoutCamInfo = False
+
+                    if EC in (4, 5):
+                        logging.debug("Received message %s from RemoteEx software." % rargs)
+                        if EC == 4 and rfunc == "Livemonitor":
+                            # TODO maybe check if rargs is empty, should not for type 4 and 5 i think
+                            # if len(rargs) > 0:
+                            self.queue_img.put(rargs)  # only put msg in queue when it notifies about an image
+
+                    else:  # send response including EC to queue
+                        self.queue_command_responses.put(msg_splitted)
+
+        except Exception:
+            logging.exception("Hamamatsu streak camera TCP/IP receiver thread failed.")
+        finally:
+            logging.info("Hamamatsu streak camera TCP/IP receiver thread ended.")
+
+    def StartAcquisition(self, AcqMode):
+        """Start an acquisition.
+        :parameter AcqMode: (str) see AcqStart
+        """
+        # Note: sync acquisition calls directly AcqStart
+
+        # TODO this is not a nice solution but seems to work for now
+        try:
+            self.AcqStart(AcqMode)
+        except Exception:  # TODO RemoteEx error not catched...
+            logging.debug("Starting acquisition currently not possible. An acquisition or live mode might be still "
+                          "running. Will stop and restart live mode.")
+            self.AcqStop()
+            timestamp = time.time()
+            timeout = 5
+            while int(self.AsyncCommandStatus()[0]):
+                time.sleep(0)
+                timestamp += time.time()
```

Comment:
If you add time.time() + time.time(), it's going to be a really large value which represent something very far in the future. That's probably not what you want.
=> Just write `if time.time() > timeout:`